packages/shared/lib/core/profile-manager/api/events/transactionProgress.ts

```diff
@@ -1,13 +1,67 @@
 import { selectedAccountId } from '@core/account'
-import { TransferProgressEventData } from '@lib/typings/events'
 import { get } from 'svelte/store'
+import { activeProfile } from '@core/profile/stores'
+import { ProfileType } from '@core/profile/enums'
+import { closePopup, openPopup } from '@lib/popup'
+import { ledgerSendConfirmationProps, ledgerDeviceStatus, ledgerMintNativeTokenProps } from '@core/ledger'
 
-export function handleTransactionProgress(accountId: string, payload: TransferProgressEventData): void {
+export function handleTransactionProgress(
+    accountId: string,
+    payload:
+        | { PreparedTransactionEssenceHash: string }
+        | { PreparedTransaction: string }
+        | 'PerformingPow'
+        | 'SigningTransaction'
+): void {
     if (get(selectedAccountId) === accountId) {
         console.warn('Transaction progress handler unimplemented: ', payload)
-        // transferState.set({
-        //     type: payload.type,
-        //     data: payload,
-        // })
+
+        if (get(activeProfile).type === ProfileType.Ledger) {
+            const _sendConfirmationProps = get(ledgerSendConfirmationProps)
+            const _mintNativeTokenProps = get(ledgerMintNativeTokenProps)
+
+            const _openPopup = (type = 'ledgerTransaction') => {
+                const shouldPreventClose = type === 'enableLedgerBlindSigning'
+
+                openPopup({
+                    type,
+                    hideClose: shouldPreventClose,
+                    preventClose: shouldPreventClose,
+                    ...(_sendConfirmationProps
+                        ? {
+                              props: {
+                                  toAddress:
+                                      _sendConfirmationProps.recipient.type === 'address'
+                                          ? _sendConfirmationProps.recipient.address
+                                          : _sendConfirmationProps.recipient.account.depositAddress,
+                                  toAmount: `${_sendConfirmationProps.amount} ${_sendConfirmationProps.unit}`,
+                                  sendConfirmationPopupProps: _sendConfirmationProps,
+                              },
+                          }
+                        : {
+                              props: {
+                                  mintNativeTokenPopupProps: _mintNativeTokenProps,
+                              },
+                          }),
+                })
+            }
+
+            if (payload === 'PerformingPow') {
+                // TODO: Should we double check the type of popup?
+                closePopup()
```

Comment:
We could write a slightly more specific function like `closeLedgerConfirmationPopup`?